tiledb/sm/buffer/const_buffer.h

```diff
@@ -55,7 +55,7 @@ class ConstBuffer {
    * @param data The data of the buffer.
    * @param size The size of the buffer.
    */
-  ConstBuffer(const void* data, uint64_t size);
+  ConstBuffer(const void* data, const uint64_t size);
```

Comment:
Do not `const` arguments in function declarations, only use it in function definitions. There's a reason for this, but most importantly to remain consistent with the rest of our code base.

More here: https://abseil.io/tips/109

"Never use top-level const on function parameters in declarations that are not definitions (and be careful not to copy/paste a meaningless const). It is meaningless and ignored by the compiler, it is visual noise, and it could mislead readers.
Do use top-level const on function parameters in definitions at your (or your team’s) discretion. You might follow the same rationale as you would for when to declare a function-local variable const."